Add procedure to sync Ubuntu ESM content (#3075)
theforeman/foreman-documentation: guides/common/modules/proc_synchronizing-ubuntu-expanded-security-maintenance-content.adoc

[id="Synchronizing-Ubuntu-Expanded-Security-Maintenance-content_{context}"]
= Synchronizing Ubuntu Expanded Security Maintenance content

Canonical provides Ubuntu https://ubuntu.com/security/esm[Expanded Security Maintenance] (ESM) repositories to their paying customers.

> Use {Project} to synchronize ESM repositories from Canonical to distribute content to your hosts running Ubuntu 14.04 and Ubuntu 16.04.

Reviewer: ```suggestion
In {Project}, synchronize ESM repositories from Canonical to distribute content to your hosts running Ubuntu 14.04 and Ubuntu 16.04.
```

> Extract the ESM repository information from an existing host running Ubuntu 14.04 or 16.04 once by hand to enable synchronization to {Project}.

Reviewer: ```suggestion
```
It isn't the extraction that enables the sync, but the "subscription attachment", probably.

Author: Removed.

[IMPORTANT]
====
Providing https://ubuntu.com/security/esm[Ubuntu ESM] repositories to hosts running Ubuntu 14.04 or 16.04 requires a subscription from Canonical.
Ensure to provide valid licenses for all used Canonical products.
Using insufficient, invalid, or otherwise inadequate licenses might violate your terms with Canonical.
====

> .One time procedure on an existing host

Reviewer: ```suggestion
.Procedure
```

Author: Applied.

> . Install `ubuntu-advantage-tools` on a host running Ubuntu 14.04 or 16.04:

Reviewer: ```suggestion
. On a host running Ubuntu 14.04 or 16.04, attach your subscription and extract repository information:
.. Install `ubuntu-advantage-tools` on a host running Ubuntu 14.04 or 16.04:
```
I would make this "one-time procedure" into the first step of a normal procedure and increase the level of steps that relate to this step (make sub-steps).

Author: Applied.

+
[source,none]
----
# apt-get install ubuntu-advantage-tools
----
. Register your host with Canonical:
+
[source,none]
----

> # ua attach ABCDEF0123456789

Reviewer: ```suggestion
# ua attach _ABCDEF0123456789_
```
User-entered string. This might require different settings for the code block.

Author: Fixed; changed for all code blocks.

----
+

> Use the code from https://canonical.com/[canonical.com].

Reviewer: ```suggestion
Use your subscription token from https://canonical.com/[canonical.com].
```

Author: Applied.

. Extract the `Upstream URL`, `Releases`, `Components`, and `Architectures` from the ESM repository:
+
[source,none]
----
# cat /etc/apt/sources.list.d/ubuntu-esm-infra.list
----
. Extract user name and password:
+
[source,none]
----
# cat /etc/apt/auth.conf.d/90ubuntu-advantage
----

> 
> .Procedure on {Project}

Reviewer: ```suggestion
```

. Add GPG public key to {Project}:

> .. Navigate to *Content* > *Content Credentials*.

Reviewer: ```suggestion
.. In the {ProjectWebUI}, navigate to *Content* > *Content Credentials*.
```

Author: Applied.

.. Click *Create Content Credential*.
.. Enter a name and the GPG public key for Ubuntu.
+
Ensure to use ASCII-armored GPG keys.
Convert the key if necessary:
+
[source,none]
----
# gpg --import ubuntu-advantage-esm-infra-trusty.gpg
# gpg --export --armor 4067E40313CB4B13
----
.. Click *Save* to save your content credential to {Project}.
. Create a product:
.. Navigate to *Content* > *Products*.
.. Click *Create Product*.

> ... Enter a *Name*, select the previously imported GPG public key, and optionally add a sync plan and description.

Reviewer: ```suggestion
... Enter a *Name*, select the previously imported GPG public key, and, optionally, add a sync plan and description.
```

... Click *Save* to save the product to {Project}.
.. Navigate to *Content* > *Products*.
.. Select the previously created product.
.. Click *New Repository*.
... Enter a *Name* and select type `deb`.
... Enter the *Upstream URL*, *Releases*, *Components*, and *Architectures* as extracted from your host from `/etc/apt/sources.list.d/ubuntu-esm-infra.list`.
... Enter the *Upstream Username* and *Upstream Password* as extracted from your host from `/etc/apt/auth.conf.d/90ubuntu-advantage`.
ifdef::orcharhino[]
... Enter *Errata URL*: `\https://dep.atix.de/dep/api/v1/ubuntu-esm`.
endif::[]
... Click *Save* to save your repository to {Project}.
. Synchronize your product.
.. Navigate to *Content* > *Products*.
.. Select the previously created product.

> .. Click *Sync Now* from the _actions_ drop down menu.

Reviewer: ```suggestion
.. In the *Actions* menu, select *Sync Now*.
```

Author: Applied.

> . Distribute and manage your content as you manage other Deb content.

Reviewer: ```suggestion

.Next steps
* Distribute and manage your content as you manage other Deb content.
```

Author: Applied.
